engine/client/secretprovider/vault.go

```diff
@@ -31,26 +55,46 @@ func vaultProvider(ctx context.Context, key string) ([]byte, error) {
 	secretPath := keyParts[0]
 	secretField := keyParts[1]
 
-	// check if client is initialized
-	if vaultClient == nil {
-		err := vaultConfigureClient(ctx)
-		if err != nil {
-			return nil, err
+	if existing, ok := vaultCache[key]; !ok || hasExpired(existing) {
+		// check if client is initialized
+		if vaultClient == nil {
+			err := vaultConfigureClient(ctx)
+			if err != nil {
+				return nil, err
+			}
 		}
-	}
 
-	// check if path is in key cache
-	if _, ok := vaultCache[key]; !ok {
 		// read the secret
 		s, err := vaultClient.KVv2(mount).Get(ctx, secretPath)
 		if err != nil {
-			return nil, err
+			return nil, fmt.Errorf("path: %q: %w", secretPath, err)
```

Comment:
With error strings, generally we only use `:` to separate each error from the next in the chain, so it's fine to do:
```suggestion
			return nil, fmt.Errorf("path %q: %w", secretPath, err)
```